rust/src/operations/optimize.rs

```diff
@@ -162,14 +218,18 @@ impl<'a> std::future::IntoFuture for OptimizeBuilder<'a> {
         let this = self;
 
         Box::pin(async move {
-            let writer_properties = this
-                .writer_properties
-                .unwrap_or_else(|| WriterProperties::builder().build());
+            let writer_properties = this.writer_properties.unwrap_or_else(|| {
+                WriterProperties::builder()
+                    .set_compression(Compression::ZSTD(ZstdLevel::try_new(4).unwrap()))
+                    .set_created_by(format!("delta-rs version {}", crate_version()))
+                    .build()
+            });
```

Comment:
Since the files are rewritten using ZSTD don't we need to update the filenames?

https://github.com/delta-io/delta-rs/blob/8288fdd4e2b51982caeffb71e2456ac779e8b7b9/rust/src/operations/writer.rs#L303

Reply:
Oh yeah. TBH I don't know why we write the compression algorithm in the filename. We don't handle this currently if a user passes in a different compression algorithm themselves.

Reply:
> TBH I don't know why we write the compression algorithm in the filename

This is again one of these relics from the early days i think. I do remember comments floating around, about figuring out the various parts in names written by spark, so I guess short answer is "b/c spark did it" 😆.

That said I do believe this to be somewhat of a "soft convention" in many parquet usages, but I may also be wrong.

Reply:
Yeah this can be fixed in a future PR. I believe it should not break anything.

Reply:
yes, this the convention. the reason behind that is it can be used as an optimization to setup the parquet reader with proper decompression decoder based on the list api response alone, without having to read the first couple of bytes from the object store to detect the compression algo.